src/TraceParserGen.Tests/ParserGenerationTests.cs

```diff
@@ -0,0 +1,288 @@
+
+            using (var process = Process.Start(startInfo))
+            {
+                var output = process.StandardOutput.ReadToEnd();
+                var error = process.StandardError.ReadToEnd();
+                
+                process.WaitForExit();
+
+                if (!string.IsNullOrWhiteSpace(output))
+                {
+                    Output.WriteLine("STDOUT:");
+                    Output.WriteLine(output);
+                }
+
+                if (!string.IsNullOrWhiteSpace(error))
+                {
+                    Output.WriteLine("STDERR:");
+                    Output.WriteLine(error);
+                }
+
+                return process.ExitCode;
+            }
+        }
+
+        private void CreateTestConsoleApp(string projectDir, string generatedParserPath)
+        {
+            // Create the .csproj file
+            string csprojContent = $@"<Project Sdk=""Microsoft.NET.Sdk"">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net462</TargetFramework>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include=""Microsoft.Diagnostics.Tracing.TraceEvent"">
+      <HintPath>{GetTraceEventAssemblyPath()}</HintPath>
+    </Reference>
+  </ItemGroup>
+</Project>";
+
+            File.WriteAllText(Path.Combine(projectDir, "TestApp.csproj"), csprojContent);
+
+            // Copy the generated parser file
+            string destParserPath = Path.Combine(projectDir, Path.GetFileName(generatedParserPath));
+            File.Copy(generatedParserPath, destParserPath, true);
+
+            // Create Program.cs that uses reflection to instantiate parsers
+            string programContent = @"using System;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Diagnostics.Tracing;
+
+class Program
+{
+    static int Main(string[] args)
+    {
+        try
+        {
+            Console.WriteLine(""Starting parser test..."");
+            
+            // Find all TraceEventParser-derived types in the current assembly
+            var assembly = Assembly.GetExecutingAssembly();
+            var parserTypes = assembly.GetTypes()
+                .Where(t => typeof(TraceEventParser).IsAssignableFrom(t) && !t.IsAbstract)
+                .ToList();
+
+            Console.WriteLine($""Found {parserTypes.Count} parser type(s)"");
+
+            foreach (var parserType in parserTypes)
+            {
+                Console.WriteLine($""  Testing parser: {parserType.Name}"");
+                
+                // Create an instance of the parser
+                // TraceEventParser constructors typically take a TraceEventSource parameter
+                // Since we don't have a real source, we'll just verify the type can be instantiated
+                // by checking if it has expected methods
+                
+                var enumerateMethod = parserType.GetMethod(""EnumerateTemplates"", 
```

Comment:
Instead of searching for `EnumerateTemplates` lets assume that the caller of this executable passed in the path to a trace file with events that will be parsed by the generated parser.  You can create a TraceEventSource by calling TraceEventDispatcher.GetDispatcherFromFileName.  Then you can make sure that the events show up properly in the parser by hooking the TraceEventParser.All on the generated parser.